glossary.md

```diff
@@ -0,0 +1,27 @@
+- external identifier - a reference to external (LIMS) data. It has a _provider_, the name of the data store that is the source for an external identifier, and an _id_ which is the store-specific name for this data. The same ID can be re-used across multiple providers.
+- external key - a reference to a specific version of external (LIMS) data. This has the same data as an _external identifier_ plus a dictionary of versions. Each version has a _version key_ and a _version value_.
+- label (analysis output) - additional information attached by the workflow run. This can serve as metadata for consumers of Vidarr output. For instance, it could be used to differentiate between read 1 and read 2 FASTQs or indicate the number of records in a file.
+- label (workflow) - additional parameters used to differentiate workflows during matching. For instance, the same input FASTQ might be aligned to two different genome references, so a label could be used to distinguish these cases.
+- metadata - information provided during launch of a workflow run to determine how the *output* should be treated. Specifically, it determines which external identifiers/keys are associated with which outputs.
+- parameters - the input information for a workflow run.
+- provider - An external data store (usually a LIMS) that holds identifiers that can be associated with _workflow runs_ and _analysis output_.
+- provisioner, input - a plugin that can use Vidarr paths or user-provided information to put a real file into a location usable by a workflow engine.
+- provisioner, output - a plugin that can take a file created by a workflow run and put it into permanent storage and provide a path, size, and MD5.
+- provisioner, runtime - a plugin that can take an identifier from a workflow engine and extract any information about the workflow itself (_e.g._, runtime performance, logs) and put them into permanent storage.
+- target - configuration of provisioners and a workflow engine that is capable of executing workflow runs.
+- workflow - a kind of analysis procedure. Each workflow can have many _versions_ that can be executed. A workflow defines _labels_ that must be included in order to execute any version. All versions of a workflow are considered equivalent for the purposes of matching (_i.e._, if a new version is available, but a successful workflow run from previous version completed, then it should not be executed again). See the _OOP analogy_ below.
+- workflow engine - an external system that can launch and monitor jobs (_e.g._, Cromwell, Nextflow).
+- workflow run - a particular execution of a workflow version. See the _OOP analogy_ below.
+- workflow version - a script that can be executed by a workflow engine that fulfills the purpose of a workflow run. For instance, there might be a workflow for _BWAmem alignment_ with multiple versions that encompass both the version of BWAmem and a WDL script used to launch it. See the _OOP analogy_ below.
```

Comment:
This looks good! I do still think 'that fulfills the purpose of a workflow run' is confusing because it sounds to me like a workflow run doesn't have its own purpose, the workflow does. Could this change to 'a script that can be executed by a workflow engine that _produces_ a workflow run'? Or something similar?

Reply:
I tried a rewrite.

Reply:
I was surprised to read that workflow runs have their own parameters, but other than that, looks good